### PR TITLE
Stg reproducible research and earlier sections

### PR DESCRIPTION
**Pull request by Deckart2**
Hi @awunderground ! This PR works on the reproducible research and programming sections of the book! I made a couple of decisions that I think are sensible, but would love your thoughts:
1. For the `06_` and `07_` sections on Quarto, I removed much of the review in `07_` because it was identical to `06_`. I also built out notes on `params` a bit
2. I moved the iteration content from `12_apis` to `10_functions` because that topic obviously deals with functions
3. I also removed the packages materials from the `10_functions` section because it is going in a new chapter. I did a simple copy paste to `11_packages` but am not rendering that right now. It is more there just to place-hold. 
4. I ran into a bug using `quarto render` it was not identifying my environment variables which caused an issue calling the census API in chapter 12. To address this, I made an [environment file](https://quarto.org/docs/projects/environment.html) with the following structure `census_api_key=12345` and added that to the gitignore. 

Let me know how this looks. Happy to make further updates as needed!

**Update May 27:** I have made tweaks to earlier sections of the book (mostly word-smithing and reorganization) that are also in this branch

### CODE REVIEW

File: 06_reproducible-research-with-quarto.qmd [hunk 1]
```diff
@@ -0,0 +1,382 @@
+---
+title: "Reproducible Research with Quarto"
+abstract: The section provides motivation for and then outlines how to use Quarto, a modern statistical programming tool and the successor of RMarkdown.
```
Comment (awunderground) [inline 06_reproducible-research-with-quarto.qmd:3]: RStudio doesn't have a space. R Markdown has a space.

File: 06_reproducible-research-with-quarto.qmd [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+An increase in travel speed of one mile per hour is associated with a `r round(model1$coefficients[2], 2)` foot increase in stopping distance on average.
+
+## Quarto
+
+Quarto is a literate statistical programming tool for R, Julia, Python, JavaScript, and more. It was released by [Posit](https://posit.co) in 2022. Quarto is an important tool for reproducible research. It combines narrative text with styles, code, and the output of code and can be used to create many types of documents including PDFs, html websites, slides, and more.
+
+Quarto builds on the success of R Markdown. In fact, Quarto will Render R Markdown (`.Rmd`) documents without any edits or changes.
+
+Sweave is a competing framework that it is out-of-date, and Jupyter (**Ju**lia, **Py**thon, and **R**) is a competing framework that is popular for Python but has not caught on for R.
```
Comment (awunderground) [inline 06_reproducible-research-with-quarto.qmd:62]: Should probably drop the Sweave reference. It's old!

Comment (awunderground) [reply 06_reproducible-research-with-quarto.qmd:62]: Maybe mention LaTeX, which is very popular.

File: 09_advanced-git.qmd [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+`git switch <new branch name>` navigates you from your current branch to the specified branch. It is only possible to switch between branches if there are no uncommitted changes. [^checkout]
+
+[^checkout]: `git checkout` is another exceedingly common git command. Many resources on the internet may encourage the usage of `git checkout <branch name>` to switch branches, or `git checkout -b <new branch name>` to create a new branch and then switch to it. This is okay! However, `git checkout` also has other capabilities, and that glut of functionality can be confusing to users. This makes `git switch` the simpler, more modern option.]
+
+
+1. Use `git switch main` to navigate to the main branch. Use `git pull origin main` to update your local version of the `main` branch with any remote changes. 
+2. Use `git switch -c <new branch name>` to create and switch to a new branch with the name `iss<number>`, where <number> is the issue number from GitHub.
+3. Work as if you are on the main branch but push to branch `iss<number>`. 
+
+[Jenny Bryan](https://happygitwithr.com/git-branches.html) provides a more thorough background.
```
Comment (awunderground) [inline 09_advanced-git.qmd:67]: I've never used `git switch`!!!

File: 11_packages.qmd [hunk 4]
```diff
@@ -0,0 +1,111 @@
+---
+title: "11_packages"
+editor: visual
+---
```
Comment (awunderground) [inline 11_packages.qmd:4]: This is a much sparser YAML header than the other chapters.

**Review by awunderground — COMMENTED** [review]
This is AWESOME! I made a few small comments. I will make the revisions and then merge!

**Review by awunderground — APPROVED** [review]

File: 09_advanced-git.qmd (hunk 3 above)
Comment (Deckart2) [reply 09_advanced-git.qmd:67]: Yes! I saw Clayton use it semi-recently (maybe in the last year) and did more research which I found enlightening.

File: 11_packages.qmd (hunk 4 above)
Comment (Deckart2) [reply 11_packages.qmd]: Yes! This is true. I commented on this in the Slack message, but the quarto document is **not** ready for primetime and is not rendered in the book. The header will 100% need to be updated.

**Pull request merged**